jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -348,6 +348,7 @@ private void processConnect(SelectionKey key, Connect connect)
                 LOG.debug("Connected {} {}", connected, channel);
             if (connected)
             {
+                _selectorManager.onConnected(channel);
```

Comment:
This call should be inside the `if` statement below, otherwise cancelling the timeout may fail and fail the connection, but we would emit the "connect success" event.